### PR TITLE
[sequelize_v4] fix: Convert module from ESM to CommonJS

### PR DESCRIPTION
**Pull request by retyui**
- Links to documentation: https://github.com/sequelize/sequelize/tree/v4/docs
- Link to GitHub or NPM: https://github.com/sequelize/sequelize
- Type of contribution: fix | refactor

Other notes:
via: https://github.com/flow-typed/flow-typed/issues/3237

### CODE REVIEW

**Comment by jedwards1211** [comment]
@pascalduez @GAntoine hey guys, maybe these defs work better for people using CommonJS, but they cause a bunch of new errors in working code for people using ES6 imports.  Really aggravating.  Can we revert this and try again to support CommonJS without breaking ES6 imports?

File: definitions/npm/sequelize_v4.x.x/flow_v0.42.x-/test_sequelize.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
 // @flow
+const Sequelize = require("sequelize");
+const { Model, Transaction } = Sequelize;
 
-import Sequelize, {Model, Transaction} from "sequelize";
```
Comment (jedwards1211) [inline definitions/npm/sequelize_v4.x.x/flow_v0.42.x-/test_sequelize.js]: @pascalduez deleting this line is unacceptable and it should have never passed review.

**Comment by jedwards1211** [comment]
@pascalduez I'd like to take on the review responsibilities for the sequelize type defs.  I'm the one who put in tons of hours to create them in the first place and keep them updated.

**Comment by retyui** [comment]
Now I managed to find out that if I use the declaration object

We can work with CommonJS and ES modules `sequelizeObject`

But if we use the class then the behavior changes 

```js
// tmpSeq_vx.x.x.js
declare module 'sequelizeObject' {
  declare module.exports: {
    Model1: number,
  };
}

declare module 'sequelizeClass' {
  declare class S {
    Model2: number;
  }

  declare module.exports: typeof S;
}

declare module 'sequelizeClass3' {
  declare class S {
    Model2: number;
  }

  declare module.exports: S;
}
//----------------------------------------------------------------------------
// test_tmpSeq.js 
import { Model1 } from 'sequelizeObject'; // Work
import S2, { 
Model2  // error "Cannot import `Model2` because there is no `Model2` export in `sequelizeClass`"
} from 'sequelizeClass'; 
import S3, { 
Model3  // work
} from 'sequelizeClass3'; 
new S2();
new S3(); // error "Cannot call `S` because `S` [1] is not a function."
```

**Comment by jedwards1211** [comment]
I'm not 100% sure what you're trying to say but doesn't that example show that none of the three ways can fully support both CommonJS and ES modules?

File: definitions/npm/sequelize_v4.x.x/flow_v0.42.x-/test_sequelize.js (hunk 1 above)
Comment (pascalduez) [reply definitions/npm/sequelize_v4.x.x/flow_v0.42.x-/test_sequelize.js]: Agree, we should test both commonjs and ES modules when doing those kind of conversions.

**Comment by pascalduez** [comment]
> @pascalduez I'd like to take on the review responsibilities for the sequelize type defs. I'm the one who put in tons of hours to create them in the first place and keep them updated.

I'm all for it, a bit of ownership will be beneficial.

**Comment by pascalduez** [comment]
> I'm assuming there isn't, which is why I don't think it's possible to support both CommonJS and ES modules

Looks like the same issue I had wit the `verror` libdef, Flow can't properly model all those cases.

**Comment by jedwards1211** [comment]
I'll watch this repo and set an email rule to filter out most other packages (there are a few others I should watch)

**Comment by jedwards1211** [comment]
or do you know any kind of bot I can set up to only watch changes to certain type defs?  Unfortunately there's no way to make a 100% perfect email filter rule...

**Comment by pascalduez** [comment]
If I'm not mistaken, TS (definitely typed) has a bot for assigning reviewers based on contributions. 
Also we should maybe require at least two approvals per PR... Might considerably bump merging time though.

**Comment by GAntoine** [comment]
@jedwards1211 You can add a CONTRIBUTING.md file to the `sequelize` folder, and mark yourself as the person to ping when there are changes. It's also a good place to detail best practices and pitfalls for that specific PR.

@pascalduez I'll add a [CODEOWNERS](https://help.github.com/en/articles/about-code-owners) file sometime this week, which will autorequest a review from specific people in certain cases.

**Pull request merged**